contracts/eosio.system/src/delegate_bandwidth.cpp

```diff
@@ -576,12 +579,22 @@ namespace eosiosystem {
             // else (remaining amount is less than genesis balance )
             // -> genesis balance is now decreased to new_delegated_to_self_tokens amount
             genesis.balance = std::min<asset>(del_bw_itr->net_weight + del_bw_itr->cpu_weight, genesis.balance);
+            withdrawn_genesis = prev_genesis - genesis.balance;
          }
        });
 
        if( owner_genesis.balance == zero_asset && owner_genesis.unclaimed_balance == zero_asset){
            genesis_tbl.erase(owner_genesis);
        }
+
+       const int64_t delta_time_usec = (gbm_final_time - current_time_point()).count();
+       const int64_t to_burn_amount = static_cast<int64_t>(withdrawn_genesis.amount * (delta_time_usec / double(useconds_in_gbm_period)));
+
+       const asset to_burn(to_burn_amount, core_symbol());
```

Comment:
Maybe we don't bother with the min(...) I mentioned above and just surround this line and the code to then end of this function with `if(to_burn_amount > 0)` ? That probably solves the after gbm_final_time case.

Reply:
Do you think it is fine if we have both validations? One related to date and the other one related to amount. I would add both...